ta/avb/entry.c

```diff
@@ -212,9 +238,117 @@ static TEE_Result write_lock_state(uint32_t pt,
 	if (count == sizeof(lock_state) && lock_state == wlock_state)
 		goto out;
 
-	res = create_state(wlock_state, &h);
+	res = create_rb_state(wlock_state, &h);
+out:
+	TEE_CloseObject(h);
+	return res;
+}
+
+static TEE_Result write_persist_value(uint32_t pt,
+				      TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	const uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			       TEE_DATA_FLAG_ACCESS_WRITE |
+			       TEE_DATA_FLAG_OVERWRITE;
+	TEE_Result res;
+	TEE_ObjectHandle h;
+
+	char *name_buf;
+	uint32_t name_buf_sz;
+
+	char name_full[TEE_OBJECT_ID_MAX_LEN];
+	uint32_t name_full_sz;
+
+	char *value;
+	uint32_t value_sz;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	name_buf = params[0].memref.buffer;
```

Comment:
You can declare these here instead and save some lines.

Reply:
I might be mistaken, but I thought that we were avoiding declarations after statements (although C99 permits doing so). Correct me if I'm wrong. Thanks

Reply:
@igoropaniuk you're wrong! ;-) https://github.com/OP-TEE/optee_os/blob/master/README.md#5-coding-standards

"Automatic variables should always be initialized. Mixed declarations and statements are allowed, and may be used to avoid assigning useless values. Please leave one blank line before and after such declarations."

So we are deviating a bit from the linux kernel rules here. This dates back from 3 months or so.
https://github.com/OP-TEE/optee_os/commit/439203cb04ce

Reply:
@jforissier missed that, thanks!